### PR TITLE
Replace debuggerflavor with debugger

### PR DESCRIPTION
**Pull request by tdesveauxPKFX**
Ref: #1060 

Remove duplicate in APIs as `debugger` can be used to set `DebuggerFlavor` in Visual Studio.

### CODE REVIEW

**Comment by TurkeyMan** [comment]
I'm still not sure I'm into this... it feels like we've confused 2 concepts now. Busy ATM, let me get back to this...

**Comment by tdesveauxPKFX** [comment]
Yeah, I was too quick in making this. I'm pretty busy as well but should be able to look into it this week end.

File: src/_premake_init.lua [hunk 1]
```diff
@@ -285,6 +285,10 @@
 			"Default",
 			"GDB",
 			"LLDB",
+			"Local",
+			"Remote",
```
Comment (TurkeyMan) [inline src/_premake_init.lua]: I don't like `Local` & `Remote` here, I think those states are implied by `debugremotehost` being set or not no?

**Comment by tdesveauxPKFX** [comment]
I somewhat forgot about this.

Just updated the branch and the only notable change was to rename Local to VisualStudioLocal (same for the others).

I think this PR can be merged. If there is further concerns about how to handle `DebuggerFlavor` for Visual Studio, there should be another issue/PR.

**Comment by starkos** [comment]
I defer to @TurkeyMan, who had some concerns about the previous versions. Will add as a reviewer.

**Comment by samsinsane** [comment]
@TurkeyMan Do you have any objections to merging this?

**Comment by starkos** [comment]
Closing this one in favor of #1357, which deprecates rather than removes `debuggerflavor()`.

**Pull request closed without merging**